torchvision/prototype/models/depth/stereo/raft_stereo.py

```diff
@@ -0,0 +1,692 @@
+from typing import List, Optional
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch import Tensor
+from torch.nn.modules.batchnorm import BatchNorm2d
+from torch.nn.modules.instancenorm import InstanceNorm2d
+from torchvision.ops import Conv2dNormActivation
+
+from torchvision.utils import _log_api_usage_once
+from torchvision.models._api import Weights, WeightsEnum
+from torchvision.models.optical_flow._utils import grid_sample, make_coords_grid, upsample_flow
+
+
+# Write helper function here temporarily
+
+# MODIFIED from torchvision.models.optical_flow._utils.upsample_flow
+# Make it more generic by adding factor as params and can handle different num_channels input
+def upsample_with_mask_and_factor(x, up_mask: Optional[Tensor] = None, factor=8):
+    """Upsample tensor x to have factor times the size
+
+    If up_mask is None we just interpolate.
+    If up_mask is specified, we upsample using a convex combination of its weights.
+    """
+    batch_size, num_channels, h, w = x.shape
+    new_h, new_w = h * factor, w * factor
+
+    if up_mask is None:
+        return factor * F.interpolate(x, size=(new_h, new_w), mode="bilinear", align_corners=True)
+
+    up_mask = up_mask.view(batch_size, 1, 9, factor, factor, h, w)
+    up_mask = torch.softmax(up_mask, dim=2)  # "convex" == weights sum to 1
+
+    upsampled_x = F.unfold(factor * x, kernel_size=3, padding=1).view(batch_size, num_channels, 9, 1, 1, h, w)
+    upsampled_x = torch.sum(up_mask * upsampled_x, dim=2)
+
+    return upsampled_x.permute(0, 1, 4, 2, 5, 3).reshape(batch_size, num_channels, new_h, new_w)
+
+
+# REUSING from torchvision.models.optical_flow.raft
```

Comment:
If you straight reuse something at this stage, it might be best to import it from raft to avoid the degree of copy-pasting and assist reviewing. We won't leave it like that at the end, but it will really help us at the early stages of the work.